File: app/src/main/java/com/norvera/confession/utils/AppExecutor.java
package com.norvera.confession.utils;

import java.util.concurrent.Executor;

public class AppExecutor {

    // TODO Make class a singleton

    private final Executor diskIO;
    private final Executor mainThread;

    public AppExecutor(Executor diskIO, Executor mainThread) {
        this.diskIO = diskIO;
        this.mainThread = mainThread;
    }

    public Executor diskIO() {
        return diskIO;
    }

    public Executor mainThread() {
        return mainThread;
    }
}

File: app/src/main/java/com/norvera/confession/utils/AppExecutor.kt
package com.norvera.confession.utils

import java.util.concurrent.Executor

class AppExecutor(
    // TODO Make class a singleton

    private val diskIO: Executor, private val mainThread: Executor
) {

    fun diskIO(): Executor {
        return diskIO
    }

    fun mainThread(): Executor {
        return mainThread
    }
}
